samples/apps/copilot-chat-app/WebApp/src/libs/semantic-kernel/SemanticKernel.ts

```diff
@@ -13,39 +15,61 @@ export class SemanticKernel {
     // eslint-disable-next-line @typescript-eslint/space-before-function-paren
     constructor(private readonly serviceUrl: string) {}
 
-    public invokeAsync = async (ask: IAsk, skillName: string, functionName: string): Promise<IAskResult> => {
-        const result = await this.getResponseAsync<IAskResult>({
-            commandPath: `skills/${skillName}/functions/${functionName}/invoke`,
-            method: 'POST',
-            body: ask,
-        });
+    public invokeAsync = async (
+        ask: IAsk,
+        skillName: string,
+        functionName: string,
+        connectorAccessToken?: string,
+    ): Promise<IAskResult> => {
+        const result = await this.getResponseAsync<IAskResult>(
+            {
+                commandPath: `skills/${skillName}/functions/${functionName}/invoke`,
+                method: 'POST',
+                body: ask,
+            },
+            connectorAccessToken,
+        );
         return result;
     };
 
-    private readonly getResponseAsync = async <T>(request: ServiceRequest): Promise<T> => {
+    private readonly getResponseAsync = async <T>(
```

Comment:
So, SK as a service is, well, a service.  As such, it requires an access token to access 

Using an ID token to authenticate is not the best in terms of security: https://auth0.com/blog/amp/id-token-access-token-what-is-the-difference/

Also, the middleware in ASP.Net will not authorize endpoints using ID tokens. I'd have to write my own authorizer to make that happen 

To authorize properly, I'll just need a basic access token (probably one with only the "access_as_user" scope)

Reply:
Changed to pass Access Token with initial scopes as defined by skScopes in Constants.ts. Will continue discussion offline about Server Auth Story